### ISSUE TITLE
SQL Script Parser - TStrings List Index Out of bounds

### ISSUE THREAD

**Issue by the-Arioch** [hgourvest/uib#10]
When a script consists of one and the only multi-line statement, that does not end with semicolon, then the parser's GetCurrentStatement tries to read the next line from after the TStrings' end - and exception is thrown.

Like such a script:
```
Create Trigger ..... on ....
before insert
as
begin
   ..... ;
   .... ;
end

```

The key points to reproduce:
1) last  (or maybe the only ) stmt in script
2) is terminated by EOF, not by semicolon

FIX:

```
function TUIBSQLParser.NextLine: boolean;
var FNextLine: integer;
begin
//  inc(FLine);  -- failes on script like "create trigger ... begin .. ; ... ; ... end" that are not ended with semicolon
//  result := (FLine < FStrings.Count);
  FNextLine := Succ(FLine);
  result := (FNextLine < FStrings.Count);
  if result then
  begin
    FLine := FNextLine;
    FStr := FStrings[FLine];
    FCursor := PChar(FStr);
  end;
end;
```

**Comment by PizzaProgram** [comment]
I've just downloaded the "Master", and this is still not fixed.

**Comment by lhengen** [comment]
The project, sadly, appears to be abandoned.  There are two pull requests, including one of my own that have not been addressed.  I tried to contact the author after a couple weeks of making the pull request and have had no response.

**Comment by ArnoBrinkman** [comment]
That is sad, i'm still using this library today in many projects.

**Comment by lhengen** [comment]
Yes the library is definitely useful, supporting both FireBird and Interbase without being tied explicitly to either one.  It's sad to see Delphi open source projects collect dust as there are not enough of them as it is.  I have attempted to contact the repo owner on LinkedIn.  Hopefully we can get the project moving again.  Let me know if you or anyone you know is interested in helping maintain the project.  Perhaps the repo owner will add others as committers.

**Comment by the-Arioch** [comment]
I am not interested above Firebird 2.5.x at the moment

**Comment by the-Arioch** [comment]
BTW, Henri's another FB-related project - the driver for Node-JS - seems top be abandoned too, as other Node-JS hacker started doing their own driver.
